jenkins/java/ql/src/WebMethodMissingPermissionCheck.md

```diff
@@ -1,9 +1,75 @@
-This Stapler web method does not perform a permission check. This can be problem if the method returns private information, or has side effects. See [the documentation](https://www.jenkins.io/doc/developer/security/form-validation/).
+This Stapler web method does not perform a permission check.
 
-### False Positive Guidance
+## Why is this a problem?
 
-This check can easily identify _false positives_, i.e. safe code that is marked as being unsafe. In that case, simply ignore this message after confirming that the code does not return private information, or have side effects.
+This is problem if the method returns private information, or has side effects.
 
-<!-- TODO More details what is 'anything sensitive' -->
+## Next Steps
 
-Additionally, this check has a known bug: It identifies methods matching the Stapler web method naming scheme (e.g. `doWhatever`) even if they have no arguments and declare a `void` return type. [Since Jenkins 2.154 and LTS 2.138.4, these methods can no longer be invoked by Stapler](https://www.jenkins.io/doc/developer/handling-requests/actions/), so do not need a permission check. In this case, mark this finding as false positive.
+<!-- Generic section used in all findings -->
+
+1. Don't panic, even if this issue is present in previous plugin releases. Do not change your code without understanding why the finding appears. You may end up not fixing the problem, only hiding it instead.
+2. Determine whether this finding is a false positive (see guidance below). This is an automated scan result, so that's always a possibility. In general, the rules err on the side of caution, so false positives are pretty common. If it is a false positive, mark it as such on the GitHub UI and you're done!
+3. If this is a true positive finding, use the documentation below to resolve it.
+
+If this finding is in a plugin hosted by the Jenkins project, you can also always [contact the Jenkins Security Team via Jira or email](https://www.jenkins.io/security/#reporting-vulnerabilities) to ask for help in resolving this finding.
+
+## Is this a false positive?
+
+This check can easily identify _false positives_, i.e. safe code that is marked as being unsafe, for any of the following reasons:
```

Comment:
Unnecessary polishing and over-engineered review comment: rewording proposal "Code patterns can easily mislead this check (_false positive_), ....."

Reply:
To me this does not work in context with the rest of the sentence?